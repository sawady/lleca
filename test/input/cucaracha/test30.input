
fun main() {
    if 1 == 1 { putChar(65) }
    if 1 == 2 { putChar(66) }
    if 2 == 2 { putChar(67) }
    if 3 == 2 { putChar(68) }
    if 3 == 3 { putChar(69) }

    x := 1
    if x == 1 { putChar(70) }
    if x == 2 { putChar(71) }
    if 1 == x { putChar(72) }
    if 2 == x { putChar(73) }
    if x == x { putChar(74) }

    x := 2
    if x == 1 { putChar(75) }
    if x == 2 { putChar(76) }
    if 1 == x { putChar(77) }
    if 2 == x { putChar(78) }
    if x == x { putChar(79) }
    putChar(10)

}
