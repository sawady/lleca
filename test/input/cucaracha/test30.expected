Cons(
  Function(
    main,
    Nil,
    Unit,
    Cons(
      StmtIf(
        ExprEq(
          ExprConstNum(
            1
          ),
          ExprConstNum(
            1
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                65
              ),
              Nil
            )
          ),
          Nil
        ),
        Nothing
      ),
      Cons(
        StmtIf(
          ExprEq(
            ExprConstNum(
              1
            ),
            ExprConstNum(
              2
            )
          ),
          Cons(
            StmtCall(
              putChar,
              Cons(
                ExprConstNum(
                  66
                ),
                Nil
              )
            ),
            Nil
          ),
          Nothing
        ),
        Cons(
          StmtIf(
            ExprEq(
              ExprConstNum(
                2
              ),
              ExprConstNum(
                2
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprConstNum(
                    67
                  ),
                  Nil
                )
              ),
              Nil
            ),
            Nothing
          ),
          Cons(
            StmtIf(
              ExprEq(
                ExprConstNum(
                  3
                ),
                ExprConstNum(
                  2
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      68
                    ),
                    Nil
                  )
                ),
                Nil
              ),
              Nothing
            ),
            Cons(
              StmtIf(
                ExprEq(
                  ExprConstNum(
                    3
                  ),
                  ExprConstNum(
                    3
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprConstNum(
                        69
                      ),
                      Nil
                    )
                  ),
                  Nil
                ),
                Nothing
              ),
              Cons(
                StmtAssign(
                  x,
                  ExprConstNum(
                    1
                  )
                ),
                Cons(
                  StmtIf(
                    ExprEq(
                      ExprVar(
                        x
                      ),
                      ExprConstNum(
                        1
                      )
                    ),
                    Cons(
                      StmtCall(
                        putChar,
                        Cons(
                          ExprConstNum(
                            70
                          ),
                          Nil
                        )
                      ),
                      Nil
                    ),
                    Nothing
                  ),
                  Cons(
                    StmtIf(
                      ExprEq(
                        ExprVar(
                          x
                        ),
                        ExprConstNum(
                          2
                        )
                      ),
                      Cons(
                        StmtCall(
                          putChar,
                          Cons(
                            ExprConstNum(
                              71
                            ),
                            Nil
                          )
                        ),
                        Nil
                      ),
                      Nothing
                    ),
                    Cons(
                      StmtIf(
                        ExprEq(
                          ExprConstNum(
                            1
                          ),
                          ExprVar(
                            x
                          )
                        ),
                        Cons(
                          StmtCall(
                            putChar,
                            Cons(
                              ExprConstNum(
                                72
                              ),
                              Nil
                            )
                          ),
                          Nil
                        ),
                        Nothing
                      ),
                      Cons(
                        StmtIf(
                          ExprEq(
                            ExprConstNum(
                              2
                            ),
                            ExprVar(
                              x
                            )
                          ),
                          Cons(
                            StmtCall(
                              putChar,
                              Cons(
                                ExprConstNum(
                                  73
                                ),
                                Nil
                              )
                            ),
                            Nil
                          ),
                          Nothing
                        ),
                        Cons(
                          StmtIf(
                            ExprEq(
                              ExprVar(
                                x
                              ),
                              ExprVar(
                                x
                              )
                            ),
                            Cons(
                              StmtCall(
                                putChar,
                                Cons(
                                  ExprConstNum(
                                    74
                                  ),
                                  Nil
                                )
                              ),
                              Nil
                            ),
                            Nothing
                          ),
                          Cons(
                            StmtAssign(
                              x,
                              ExprConstNum(
                                2
                              )
                            ),
                            Cons(
                              StmtIf(
                                ExprEq(
                                  ExprVar(
                                    x
                                  ),
                                  ExprConstNum(
                                    1
                                  )
                                ),
                                Cons(
                                  StmtCall(
                                    putChar,
                                    Cons(
                                      ExprConstNum(
                                        75
                                      ),
                                      Nil
                                    )
                                  ),
                                  Nil
                                ),
                                Nothing
                              ),
                              Cons(
                                StmtIf(
                                  ExprEq(
                                    ExprVar(
                                      x
                                    ),
                                    ExprConstNum(
                                      2
                                    )
                                  ),
                                  Cons(
                                    StmtCall(
                                      putChar,
                                      Cons(
                                        ExprConstNum(
                                          76
                                        ),
                                        Nil
                                      )
                                    ),
                                    Nil
                                  ),
                                  Nothing
                                ),
                                Cons(
                                  StmtIf(
                                    ExprEq(
                                      ExprConstNum(
                                        1
                                      ),
                                      ExprVar(
                                        x
                                      )
                                    ),
                                    Cons(
                                      StmtCall(
                                        putChar,
                                        Cons(
                                          ExprConstNum(
                                            77
                                          ),
                                          Nil
                                        )
                                      ),
                                      Nil
                                    ),
                                    Nothing
                                  ),
                                  Cons(
                                    StmtIf(
                                      ExprEq(
                                        ExprConstNum(
                                          2
                                        ),
                                        ExprVar(
                                          x
                                        )
                                      ),
                                      Cons(
                                        StmtCall(
                                          putChar,
                                          Cons(
                                            ExprConstNum(
                                              78
                                            ),
                                            Nil
                                          )
                                        ),
                                        Nil
                                      ),
                                      Nothing
                                    ),
                                    Cons(
                                      StmtIf(
                                        ExprEq(
                                          ExprVar(
                                            x
                                          ),
                                          ExprVar(
                                            x
                                          )
                                        ),
                                        Cons(
                                          StmtCall(
                                            putChar,
                                            Cons(
                                              ExprConstNum(
                                                79
                                              ),
                                              Nil
                                            )
                                          ),
                                          Nil
                                        ),
                                        Nothing
                                      ),
                                      Cons(
                                        StmtCall(
                                          putChar,
                                          Cons(
                                            ExprConstNum(
                                              10
                                            ),
                                            Nil
                                          )
                                        ),
                                        Nil
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  ),
  Nil
)
